src/Microsoft.Azure.SignalR/DependencyInjectionExtensions.cs

```diff
@@ -27,7 +30,19 @@ public static class AzureSignalRDependencyInjectionExtensions
         /// </remarks>
         public static ISignalRServerBuilder AddAzureSignalR(this ISignalRServerBuilder builder)
         {
+
             builder.Services.AddSingleton<IConfigureOptions<ServiceOptions>, ServiceOptionsSetup>();
+
+#if NETCOREAPP3_0
+            var serviceOptions = builder.Services
```

Comment:
What is this?

Reply:
Build the configurations, so it could fallback to local SignalR with the flag Azure:SignalR:Enabled = false configuration.

Reply:
We shouldn't be building the service provider before it's ready. This isn't the right place to do this.

Reply:
I'd prefer to get the configuration in an early stage before unused dependencies add and build. Would use ConfigurationBuilder() to load appsettings.json directly better?

Reply:
Will revert this part and split to a separate PR.